### PR TITLE
My profile/tc 08 01/profile dropdown menu

### PR DESCRIPTION
**Pull request by NataliaKengil**
[TC_08_01 | Verify the Profile Dropdown Menu is displayed on "My Profile" Button Click
](https://docs.google.com/document/d/1ff46TvvPYvH_AZfc4Ilqv6Dbo9Jxp31GQh-IJhrr0wc/edit?tab=t.0#heading=h.gu0m5ch4yg2x)

### CODE REVIEW

File: pom/components/HeaderComponent.js [hunk 1]
```diff
@@ -3,8 +3,14 @@ import { step } from 'allure-js-commons';
 export default class HeaderComponent {
     constructor(page) {
         this.page = page;
-
         this.myProfileButton = this.page.getByRole('button', { name: 'My profile' });
+        this.myProfileDropdownMenu = this.page.locator('ul.menu-list_menu-list-items__gM9KR');
```
Comment (annaafanaseva) [inline pom/components/HeaderComponent.js:7]: better not to use dynamically generated class name.. consider using partial class name (without 'gM9KR')

File: pom/components/HeaderComponent.js [hunk 2]
```diff
@@ -3,8 +3,14 @@ import { step } from 'allure-js-commons';
...
+        this.myProfileDropdownMenuItems = this.page.locator('li.menu-item_menu-item__sy6_P');
```
Comment (annaafanaseva) [inline pom/components/HeaderComponent.js:8]: same here

File: pom/components/HeaderComponent.js [hunk 3]
```diff
@@ -3,8 +3,14 @@ import { step } from 'allure-js-commons';
...
+        this.myProfileDropdownMenuAccountSettings = this.page.getByRole('button', { name: 'Account settings' });
+        this.myProfileDropdownMenuBillling = this.page.getByRole('button', { name: 'Billing' });
+        this.myProfileDropdownMenuCurrency = this.page.getByRole('button', { name: 'BCurrency USD ($)' });
```
Comment (annaafanaseva) [inline pom/components/HeaderComponent.js:11]: double-check button name?

File: pom/components/HeaderComponent.js [hunk 4]
```diff
@@ -3,8 +3,14 @@ import { step } from 'allure-js-commons';
 export default class HeaderComponent {
     constructor(page) {
         this.page = page;
-
         this.myProfileButton = this.page.getByRole('button', { name: 'My profile' });
+        this.myProfileDropdownMenu = this.page.locator('ul.menu-list_menu-list-items__gM9KR');
+        this.myProfileDropdownMenuItems = this.page.locator('li.menu-item_menu-item__sy6_P');
+        this.myProfileDropdownMenuAccountSettings = this.page.getByRole('button', { name: 'Account settings' });
```
Comment (annaafanaseva) [inline pom/components/HeaderComponent.js:9]: we also have _accountSettingsLink_ below - seems to be the same element.. need to leave one

File: testData.js [hunk 5]
```diff
@@ -81,6 +81,14 @@ export const WHOIS_SEARCH_RESULT_TITLES = [
     'Registrar URL',
 ];
 
+export const DROPDOWN_ITEMS = [
```
Comment (annaafanaseva) [inline testData.js:84]: maybe add 'MYPROFILE_' to the name?

File: tests/08MyProfile.spec.js [hunk 6]
```diff
@@ -1,10 +1,53 @@
...
+        await loginUser(page, headerComponent, loginPage);
+        await page.waitForURL(process.env.URL);
+
+        await headerComponent.clickMyProfileButton();
+
+        await step('Verify the Profile Dropdown menu is displayed and contain 5 items', async () => {
+            await expect(headerComponent.myProfileDropdownMenu).toBeVisible();
+            await expect(headerComponent.myProfileDropdownMenuItems).toHaveCount(5);
+
+            const dropdownlist = await headerComponent.myProfileDropdownMenuItems.allInnerTexts();
```
Comment (annaafanaseva) [inline tests/08MyProfile.spec.js:29]: maybe replace with shorter check for items - toHaveText(DROPDOWN_ITEMS)?

Comment (NataliaKengil) [reply tests/08MyProfile.spec.js]: Анна, все ваши комментария учла и исправила.

File: tests/08MyProfile.spec.js [hunk 7]
```diff
@@ -1,10 +1,49 @@
...
 test.describe('My profile', () => {
+    test('TC_08_01 | Verify the Profile Dropdown Menu is displayed on "My Profile" Button Click', async ({
+        page,
+        loginPage,
+        headerComponent,
+    }) => {
+        await tags('My profile', 'Positive');
+        await severity('normal');
+        await description('To verify the Profile Dropdown Menu on clicking "My Profile');
+        await issue(`${QASE_LINK}previewMode=side&suite=10&tab=&case=24`, 'My profile');
```
Comment (annaafanaseva) [inline tests/08MyProfile.spec.js:16]: the link can be shorter:
await issue(`${QASE_LINK}/01-24`

yours is missing smth and returns 404

**Review by annaafanaseva — APPROVED** [review]

**Review by darin4 — APPROVED** [review]
LGTM

**Pull request merged**